insights-notification-schemas-java/src/main/java/com/redhat/cloud/notifications/validator/LocalDateTimeValidator.java

```diff
@@ -0,0 +1,32 @@
+package com.redhat.cloud.notifications.validator;
+
+import com.networknt.schema.Format;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class LocalDateTimeValidator implements Format {
+
+    private String message;
+
+    @Override
+    public String getName() {
+        return "date-time";
+    }
+
+    @Override
+    public boolean matches(String s) {
+        try {
+            DateTimeFormatter.ISO_DATE_TIME.parse(s);
+            return true;
+        } catch (DateTimeParseException exception) {
```

Comment:
Shouldn't this catch all kinds of exceptions? `DateTimeFormatter#parse` can throw a `RuntimeException` from what I can see in its source.

Reply:
From what I saw (maybe reading old java docs?) I didn't see a `RuntimeException`. I'm guessing it will fail the whole parsing. will check.

https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#parse-java.lang.CharSequence-

Reply:
From what I have seen in the code of the openjdk, it [catches](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/time/format/DateTimeFormatter.java#L1877) all the `RuntimeException`s and [throws a DateTimePArseException](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/time/format/DateTimeFormatter.java#L2017)

Reply:
My bad, I read the sources too fast. It only throws `DateTimeParseException` indeed.